server/src/main/java/org/candlepin/model/ConsumerCurator.java

```diff
@@ -1338,4 +1338,24 @@ public int cullInvalidConsumerContentAccess(Owner owner, String... existingModes
         return updated;
     }
 
+    /**
+     * Fetches all unique service type values set by all the consumers of the specified owner.
+     *
+     * @param owner
+     *  The owner the consumers belong to.
+     * @return
+     *  A list of the all the distinct values of the service types that the consumers belonging to the
+     *  specified owner have set.
+     */
+    @SuppressWarnings("unchecked")
+    @Transactional
+    public List<String> getDistinctSyspurposeServiceTypeByOwner(Owner owner) {
+        return this.createSecureCriteria()
```

Comment:
I don't believe we really need the secure criteria. The API endpoint this is called from already has a `@Verify(Owner.class)` and that should be enough I think, so this could be re-written in non-secure JPA criteria.

Also, this is a good opportunity to merge the other 3 similar methods that fetch other syspurpose attributes (`getDistinctSyspurposeRolesByOwner`, `getDistinctSyspurposeUsageByOwner`, `getDistinctSyspurposeServicelevelByOwner`) with this one, into a single one that takes a parameter, to avoid duplicating code.

Reply:
Added a `getDistinctSyspurposeValuesByOwner(Owner owner, String sysPurposeAttribute)` method to fetch syspurpose attributes based on input parameter.

Reply:
ack from me